src/lightning/app/core/plugin.py

```diff
@@ -23,77 +25,41 @@
 from pydantic import BaseModel
 
 from lightning.app.utilities.app_helpers import Logger
-from lightning.app.utilities.cloud import _get_project
 from lightning.app.utilities.component import _set_flow_context
 from lightning.app.utilities.enum import AppStage
-from lightning.app.utilities.network import LightningClient
+from lightning.app.utilities.load_app import _load_plugin_from_file
 
 logger = Logger(__name__)
 
 
-class Plugin:
-    """A ``Plugin`` is a single-file Python class that can be executed within a cloudspace to perform actions."""
+class LightningPlugin:
+    """A ``LightningPlugin`` is a single-file Python class that can be executed within a cloudspace to perform
+    actions."""
 
     def __init__(self) -> None:
-        self.app_url = None
+        self.project_id = None
+        self.cloudspace_id = None
+        self.cluster_id = None
 
-    def run(self, name: str, entrypoint: str) -> None:
-        """Override with the logic to execute on the client side."""
+    def run(self, *args: str, **kwargs: str) -> None:
+        """Override with the logic to execute on the cloudspace."""
 
-    def run_app_command(self, command_name: str, config: Optional[BaseModel] = None) -> Dict[str, Any]:
-        """Run a command on the app associated with this plugin.
+    def run_job(self, name: str, app_entrypoint: str, env_vars: Optional[Dict[str, str]] = None) -> None:
```

Comment:
```suggestion
    def run_job(self, name: str, app_entrypoint: str, env_vars: Optional[Dict[str, str]] = None, **kwargs) -> None:
```

More elements might be added like secrets, etc..

Reply:
Yes, these can be added to the dispatch as needed, will leave for follow-up 😃